test/expectations/expectations.go

```diff
@@ -83,6 +84,18 @@ func ExpectApplied(ctx context.Context, c client.Client, objects ...client.Objec
 	}
 }
 
+func ExpectObjectStatus(ctx context.Context, c client.Client, obj status.Object, conditions ...status.Condition) {
```

Comment:
```suggestion
func ExpectStatusCondition(ctx context.Context, c client.Client, obj status.Object, conditions ...status.Condition) {
```

Reply:
Did ExpectStatus() since we can have zero, one, or multiple status conditions.